tests/Fixtures/Integration/misc/PHP8_0.test

```diff
@@ -0,0 +1,367 @@
+// no syntax changes
+
+// https://wiki.php.net/rfc/str_contains
+// no syntax changes
+
+// https://wiki.php.net/rfc/deprecate_curly_braces_array_access
+// syntax was deprecated, nothing we can do - a `normalize_index_brace` rule already exists and can be used on PHP <8 (or PHP 8 if errors are ignored)
+
+// https://wiki.php.net/rfc/concatenation_precedence
+// nothing we can do
+
+// https://wiki.php.net/rfc/inheritance_private_methods
+// nothing we can do
+
+// https://wiki.php.net/rfc/ternary_associativity
+// nothing we can do about the warning - a rule can be implemented to run on PHP <8 (or PHP 8 if errors are ignored)
+
+// https://wiki.php.net/rfc/stringable
+// nothing we can do
+
+// Date: mktime() and gmmktime() now require at least one argument. time() can be used to get the current timestamp.
+time();
+time();
+mktime($a);
+gmmktime(1, 2, 3, 4, 5, 6);
+
+// Exif: Removed read_exif_data(). exif_read_data() should be used instead.
+exif_read_data($filename, $sections_needed, $sub_arrays, $read_thumbnail);
+
+// The imap_header() function which is an alias of imap_headerinfo() has been removed.
+imap_headerinfo();
+
+// parse_str() can no longer be used without specifying a result array.
+// nothing we can do
+
+// Calling implode() with parameters in a reverse order ($pieces, $glue) is no longer supported.
+implode('', $pieces);
+implode('', $pieces);
+
+// Mbstring A number of deprecated mbregex aliases have been removed.
+// nothing we can do
+
+// Namespaced names can no longer contain whitespace (nor comments)
+// nothing we can do
+
+--INPUT--
+<?php
+
+// https://wiki.php.net/rfc/named_params
+array_fill(start_index: 0, num: 100, value: 50);
+htmlspecialchars($string, double_encode: false);
+array_slice($array, $offset, $length, preserve_keys: true);
+new ParamNode($name, variadic: $isVariadic, byRef: $passByRef);
+array_foobar(array: $value); // reserved keyword as parameter name!
+
+// https://wiki.php.net/rfc/nullsafe_operator
+// TODO - CRASHING !!!
+// $country = $session?->user?->getAddress()?->country;
+
+// https://wiki.php.net/rfc/attributes_v2
+// https://wiki.php.net/rfc/attribute_amendments
+// https://wiki.php.net/rfc/shorter_attribute_syntax
+// https://wiki.php.net/rfc/shorter_attribute_syntax_change
+// TODO - BREAKING CODE !!!
+/*
+#[MyAttribute]
+#[\MyExample\MyAttribute]
+#[MyAttribute(1234)]
+#[MyAttribute(value: 1234)]
+#[MyAttribute(MyAttribute::VALUE)]
+#[MyAttribute(array("key" => "value"))]
+#[MyAttribute(100 + 200)]
+#[Attribute(Attribute::TARGET_METHOD | Attribute::TARGET_FUNCTION)]
+#[MyAttribute(1234), MyAttribute(5678)]
+class PostsController
+{
+    #[Route("/api/posts/{id}", methods: ["GET"])]
+    public function get($id): void
+    {
+    }
+}
+
+$object = new #[ExampleAttribute] class ()
+{
+    #[ExampleAttribute]
+    public function foo(#[ExampleAttribute] $bar): void
+    {
+    }
+};
+*/
+
+// https://wiki.php.net/rfc/union_types_v2
+// TODO - CRASHING !!!
```

Comment:
@SpacePossum , can you help with this crashing code sample?

Reply:
look into it, first part is https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5405

Reply:
@SpacePossum , #5405 done the fix for method params, but not for class properties. can you take a look?